onfleet/onfleet.py

```diff
@@ -168,14 +171,27 @@ def __call__(self, *args, **kwargs):
             json_response = response.json()
 
             if 'code' in json_response:
-                message = json_response['message']
-                cause = message.get('cause', '')
-
-                if isinstance(cause, dict) and cause['type'] == 'duplicateKey':
-                    raise OnfleetDuplicateKeyException("{}: {} (value: '{}', key: '{}')" \
-                        .format(message['error'], message['message'], cause['value'], cause['key']))
-                raise OnfleetException("{}: {} ({})" \
-                    .format(message['error'], message['message'], cause))
+                # So presumably this is an error response,
+                # Map these to better python variable names, since onfleet uses
+                # horrible naming conventions
+                error_type = json_response['code']
+                error_data = json_response['message']
+                error_cause = error_data['cause']
+                error_code = error_data['error']
+                error_message = error_data['message']
+
+                error_string = '{code}: {message} (cause)'.format(
+                    code=error_code,
+                    message=error_message,
+                    cause=error_cause,
+                )
+
+                if error_type == 'duplicateKey':
+                    raise OnfleetDuplicateKeyException(error_string)
+                elif error_cause == 'Geocoding errors found.':
```

Comment:
Is there really no better way to find geocoding errors than to match on a human-readable string? They don't have a specific error code or http status? 

If that's true, I'm pretty uncomfortable about this — it feels super super brittle.